files/zh-cn/web/javascript/reference/global_objects/promise/index.md

```diff
@@ -99,148 +115,135 @@ const promiseC = promiseA.then(handleFulfilled2, handleRejected2);
+      then(onFulfilled, onRejected) {
+        onFulfilled(42);
+      },
+    });
+  },
+};
+
+Promise.resolve(aThenable); // 一个兑现值为 42 的 Promise
 ```
 
-同样的概念也适用与 promise。如果我们稍加修改上面的示例，我们就能得到这个：
+### Promise 并发
 
-```html
-<!DOCTYPE html>
-<iframe></iframe> <!-- we have a realm here -->
-<script> // we have a realm here as well
-  const bound = frames[0].postMessage.bind(
-    frames[0], "some data", "*");
-    // bound is a built in function -- there is no user
-    // code on the stack -- which realm do we use?
-  Promise.resolve(undefined).then(bound);
-  // this still works, because we use the youngest
-  // realm (the incumbent) on the stack
-</script>
-```
+`Promise` 类提供了四个静态方法来促进异步任务的[并发](https://zh.wikipedia.org/wiki/并发计算)：
 
-如果我们修改代码，使用文档中的 `<iframe>` 来监听发送的消息，我们可以观察到现有设置对象的影响：
+- {{jsxref("Promise.all()")}}
+  - : `Promise.all` 方法返回一个 Promise，当**所有**传入的 Promise 都被兑现时，该 Promise 也成功兑现；当**任意一个** Promise 被拒绝时，该 Promise 也被拒绝。
+- {{jsxref("Promise.allSettled()")}}
+  - : 当**所有**的 Promise 都被敲定时（不论是被兑现还是被拒绝），`Promise.allSettled` 方法返回一个已兑现的 Promise。
+- {{jsxref("Promise.any()")}}
+  - : 当**任意一个** Promise 被兑现时，`Promise.any` 方法返回一个成功兑现的 Promise；当**所有**的 Promise 都被拒绝时，`Promise.any` 方法返回一个被拒绝的 Promise。
+- {{jsxref("Promise.race()")}}
+  - : 当**任意一个** Promise 被敲定时（无论是被兑现还是被拒绝），`Promise.race` 方法返回一个已兑现的 Promise。换句话说，当**任意一个** Promise 被兑现时，返回一个成功兑现的 Promise；当**任意一个**的 Promise 被拒绝时，返回一个被拒绝的 Promise。
 
-```html
-<!-- y.html -->
-<!DOCTYPE html>
-<iframe src="x.html"></iframe>
-<script>
-  const bound = frames[0].postMessage.bind(frames[0], "some data", "*");
-  Promise.resolve(undefined).then(bound);
-</script>
-```
-
-```html
-<!-- x.html -->
-<!DOCTYPE html>
-<script>
-window.addEventListener("message", (event) => {
-  document.querySelector("#text").textContent = "hello";
-  // 这一部分代码仅在追踪现有设置对象的浏览器中会被运行
-  console.log(event);
-}, false);
-</script>
-```
+所有这些方法都接受一个 Promise（确切地说是 [thenable](#thenable)）的[可迭代对象](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)，并返回一个新的 Promise。它们都支持子类化，这意味着它们可以在 `Promise` 的子类上调用，结果将是一个属于子类类型的 Promise。为此，子类的构造函数必须实现与 [`Promise()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) 构造函数相同的签名——接受一个以 `resolve` 和 `reject` 回调函数作为参数的单个 `executor` 函数。子类还必须有一个 `resolve` 静态方法，可以像 {{jsxref("Promise.resolve()")}} 一样调用，以将值解析为 Promise。
 
-在上面的示例中，`<iframe>` 仅在现有设置对象被追踪时才会被更新。这是因为在不追踪的情况下，我们可能会使用错误的环境发送消息。
-
-> **备注：** 目前，Firefox 完全实现了现有领域追踪，Chrome 和 Safari 仅部分实现。
+请注意，JavaScript 的本质上是[单线程的](/zh-CN/docs/Glossary/Thread)，因此在任何时刻，只有一个任务会被执行，尽管控制权可以在不同的 Promise 之间切换，从而使 Promise 的执行看起来是并发的。在 JavaScript 中，[并行执行](https://zh.wikipedia.org/wiki/并行计算)只能通过[ worker 线程](/zh-CN/docs/Web/API/Web_Workers_API)实现。
 
 ## 构造函数
 
 - {{jsxref("Promise/Promise", "Promise()")}}
-  - : 创建一个新的 `Promise` 对象。该构造函数主要用于包装还没有添加 promise 支持的函数。
+  - : 创建一个新的 `Promise` 对象。该构造函数主要用于封装还没有添加 promise 支持的函数。
+
+## 静态属性
+
+- {{jsxref("Promise/@@species", "Promise[@@species]")}}
+  - : 返回用于构造从 Promise 方法返回值的构造函数。
 
 ## 静态方法
```

Comment:
Still need updates on this section, please refer mdn/content#27560, which have made some more changes :)

Reply:
OK, I will mark this PR as a draft until everything was done.